Formula/git-flow.rb

```diff
@@ -42,13 +42,18 @@ class GitFlow < Formula
   def install
     system "make", "prefix=#{libexec}", "install"
     bin.write_exec_script libexec/"bin/git-flow"
-
     resource("completion").stage do
       bash_completion.install "git-flow-completion.bash"
       zsh_completion.install "git-flow-completion.zsh"
     end
   end
 
+  def post_install
+    cp("#{zsh_completion}/git-flow-completion.zsh", "#{zsh_completion}/_git-flow")
+    inreplace "#{zsh_completion}/_git-flow", "#!zsh", "#compdef git-flow"
+    (HOMEBREW_PREFIX/"share/zsh/site-functions").install_symlink zsh_completion/"_git-flow"
+  end
```

Comment:
I think we can get rid of all of this but the `inreplace`, which you can do in the `install` method.

Reply:
1. The install method is ignored if it's using bottles.
2. The file needs to be called `"_git-flow"`
3. Renaming the file alone would break for existing users.

Reply:
Removing the bottles, and keeping it in the source makes sense to me. 
But keep in mind, renaming the original file will break things for people who are currently sourcing it. It may be the wrong approach but the file itself mentions to do so, so can't blame users for it.

Reply:
The bottles will be rebuilt when this PR is merged.

Reply:
@carlocab Ok, that works for me, but it will break things for existing users, is that fine?

Reply:
The bottles being rebuilt, will keep the same hash, or do I need to do something?

Reply:
It's easy to fix, doesn't happen silently, and happens in an interactive context so users won't really miss it. Seems fine. If you're that concerned you can still a `git-flow-completion.sh` symlink. And/or document the change in the caveats.

Reply:
Nah, I think @SMillerDev makes a good point of `But it will force homebrew to ship a workaround for eternity,`
Lets rip the band aid. 

I pushed the changes suggested.